### ISSUE TITLE
Switching goals

### ISSUE THREAD

**Issue by bhairavmehta95** [david-abel/simple_rl#49]
What's the easiest way to switch an MDP, while still using the same agent? Can I just call `run_experiments` twice, with different MDPs but the same agent?

Context: I'm looking to see how fast an agent can adapt, when I switch the goal of the GridworldMDP.

**Comment by david-abel** [comment]
There's one built in way to do this but it might not support the precise experiment you're looking for. There is a built in [MDPDistribution](https://github.com/david-abel/simple_rl/blob/master/simple_rl/mdp/MDPDistributionClass.py) class that expects a dictionary of MDP-probability key-value pairs. Then, instead of using run_experiments, use the [run_agents_lifelong](https://github.com/david-abel/simple_rl/blob/master/simple_rl/run_experiments.py)() function (see line 122). This will run an experiment where at the end of each episode, a new MDP is sampled and the given agent learns on the new MDP.

In short: take a look at the [lifelong_example](https://github.com/david-abel/simple_rl/blob/master/examples/lifelong_example.py) for a quick way of running something like this. Hope this helps!

Beyond that, you would likely need to implement a new experiment wrapper that makes multiple calls to [run_single_agent_on_mdp](run_single_agent_on_mdp) (see line 304) without resetting the agent.

**Comment by bhairavmehta95** [comment]
Ah gotcha; perfect thank you!
